### PR TITLE
Rdcc 2481 invalid idam roles

### PR DESCRIPTION
**Pull request by nitishsingla1234**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDCC-2481

### Change description ###
Show a valid message if the idam roles provided in excel are not valid.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/CaseWorkerServiceImpl.java [hunk 1]
```diff
@@ -508,20 +509,29 @@ public CaseWorkerProfile mapCaseWorkerProfileRequest(String idamId,
...
-                clazz = UserProfileCreationResponse.class;
-            } else {
-                clazz = ErrorResponse.class;
-                validationServiceFacade.logFailures(UP_CREATION_FAILED, cwrdProfileRequest.getRowId());
+
+            clazz = (response.status() == 201 || response.status() == 409)
+                    ? UserProfileCreationResponse.class : ErrorResponse.class;
+
+            responseEntity = JsonFeignResponseUtil.toResponseEntity(response, clazz);
+            if (clazz == ErrorResponse.class && nonNull(responseEntity.getBody())) {
```
Comment (shreedhar-hmcts) [inline src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/CaseWorkerServiceImpl.java:524]: please check if body comes null with status code 4XX, will it log in exception table ?

Comment (SujeethKumarAri) [inline src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/CaseWorkerServiceImpl.java:524]: I think you need to write unit test cases for these changes - https://sonarcloud.io/component_measures?id=uk.gov.hmcts.reform%3Areform%3A%3Ard-caseworker-ref-api&metric=new_coverage&pullRequest=230&selected=uk.gov.hmcts.reform%3Areform%3A%3Ard-caseworker-ref-api%3Asrc%2Fmain%2Fjava%2Fuk%2Fgov%2Fhmcts%2Freform%2Fcwrdapi%2Fservice%2Fimpl%2FCaseWorkerServiceImpl.java&view=list

**Review by SujeethKumarAri — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/CaseWorkerServiceImpl.java (hunk 1 above)
Comment (nitishsingla1234) [reply src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/CaseWorkerServiceImpl.java]: Added the test case, which bumps up the coverage.

Comment (nitishsingla1234) [reply src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/CaseWorkerServiceImpl.java]: Updated the code to handle this scenario, and it is working as expected.

Comment (SujeethKumarAri) [reply src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/CaseWorkerServiceImpl.java]: Thanks much @nitishsingla1234 . Coverage has been improved drastically :)

**Review by SujeethKumarAri — DISMISSED** [review]

**Review by kavithagade12 — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/CaseWorkerServiceImpl.java [hunk 2]
```diff
@@ -508,20 +509,29 @@ public CaseWorkerProfile mapCaseWorkerProfileRequest(String idamId,
...
-                clazz = UserProfileCreationResponse.class;
-            } else {
-                clazz = ErrorResponse.class;
-                validationServiceFacade.logFailures(UP_CREATION_FAILED, cwrdProfileRequest.getRowId());
+
+            clazz = (response.status() == 201 || response.status() == 409)
+                    ? UserProfileCreationResponse.class : ErrorResponse.class;
+
+            responseEntity = JsonFeignResponseUtil.toResponseEntity(response, clazz);
+            if (clazz == ErrorResponse.class) {
```
Comment (abhijitdiwan) [inline src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/CaseWorkerServiceImpl.java:524]: responseEntity.getBody() we can collect it  will reduce no calls 3-4 times

Object response = responseEntity.getBody();

Comment (nitishsingla1234) [reply src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/CaseWorkerServiceImpl.java:524]: Updated the code to use a local variable.

**Review by abhijitdiwan — APPROVED** [review]

**Review by SujeethKumarAri — APPROVED** [review]

**Pull request merged**